src/core/dynamic_batch_scheduler.cc

```diff
@@ -150,17 +150,55 @@ DynamicBatchScheduler::Enqueue(std::unique_ptr<InferenceRequest>& request)
       request->Trace(), TRITONSERVER_TRACE_QUEUE_START,
       request->QueueStartNs());
 
+  // Response cache helpers
+  bool response_cache_enabled_ = model_->Server()->ResponseCacheEnabled();
+  uint64_t request_hash = 0U;
+  bool cache_hit = false;
+
   if (!dynamic_batching_enabled_) {
-    if (preserve_ordering_) {
-      DelegateResponse(request);
+    // TODO: Move cache lookup logic to a helper method
+    std::unique_ptr<InferenceResponse> cached_response;
```

Comment:
This can probably be cleaned up by implementing a `ResponseCacheManager` to keep track of cache state and join some of this logic into a helper method for hash, lookup, etc.